compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1999,6 +1999,9 @@ impl BorrowKind {
 }
 
 impl BinOp {
+    /// The checkable operators are those whose overflow checking behavior is controlled by
+    /// -Coverflow-checks option. The remaining operators have either no overflow conditions (e.g.,
+    /// BitAnd, BitOr, BitXor) or are always checked for overflow (e.g., Div, Rem).
```

Comment:
UnOp does not have is_checkable. Given these docs, that seems surprising? `Neg` behaves like the 'checkable' bin ops, doesn't it?

Reply:
The surprising part is that there is no such function on `UnOp` -- that indicates some asymmetry in how UnOp vs BinOp are handled.

Should we also have this on UnOp, and then replace the existing direct matching on `Neg` that I assume is happening with an is_checkable call?

Reply:
What does this have to do with assertion kinds? I was thinking of MIR building.

Though with https://github.com/rust-lang/rust/pull/108282, MIR building does not use this function any more anyway. Then it's only the codegen backends and yeah there it has to do with assertion kinds.